cmd/snap-repair/runner.go

```diff
@@ -285,11 +287,140 @@ func (run *Runner) initState() error {
 	}
 	// best-effort remove old
 	os.Remove(dirs.SnapRepairStateFile)
-	// TODO: init Device
+	if err := run.initDeviceInfo(); err != nil {
+		return err
+	}
 	run.stateModified = true
 	return run.SaveState()
 }
 
+func trustedBackstore(trusted []asserts.Assertion) asserts.Backstore {
+	trustedBS := asserts.NewMemoryBackstore()
+	for _, t := range trusted {
+		trustedBS.Put(t.Type(), t)
+	}
+	return trustedBS
+}
+
+func checkAuthorityID(a asserts.Assertion, trusted asserts.Backstore) error {
+	assertType := a.Type()
+	if assertType != asserts.AccountKeyType && assertType != asserts.AccountType {
+		return nil
+	}
+	// check that account and account-key assertions are signed by
+	// a trusted authority
+	acctID := a.AuthorityID()
+	_, err := trusted.Get(asserts.AccountType, []string{acctID}, asserts.AccountType.MaxSupportedFormat())
+	if err != nil && err != asserts.ErrNotFound {
+		return err
+	}
+	if err == asserts.ErrNotFound {
```

Comment:
Sorry, I forgot to remove this comment. Please ignore it. The pattern is used across the whole file and in other cases it improves readability. Seems good as it is.